frontend/public/components/factory/list.tsx

```diff
@@ -325,6 +326,22 @@ export const Rows: React.SFC<RowsProps> = (props) => {
   </div>;
 };
 
+VirtualRows.propTypes = {
+  data: PropTypes.arrayOf(PropTypes.object),
+  expand: PropTypes.bool,
+  Row: PropTypes.func.isRequired,
+};
+
+const Rows: React.SFC<RowsProps> = (props) => {
+  const {Row, expand, kindObj} = props;
+
+  return <div className="co-m-table-grid__body">
+    { props.data.map((obj, i) => <div key={i} className="co-m-row">
```

Comment:
Better to use UID if we can.

```suggestion
    { props.data.map((obj, i) => <div key={_.get(obj, 'metadata.uid', i)} className="co-m-row">
```